src/sentry/static/sentry/app/views/organizationGroupDetails/groupMerged/mergedList.tsx

```diff
@@ -1,37 +1,53 @@
 import React from 'react';
 import styled from '@emotion/styled';
-import PropTypes from 'prop-types';
 
 import EmptyStateWarning from 'app/components/emptyStateWarning';
 import Pagination from 'app/components/pagination';
 import {Panel} from 'app/components/panels';
 import QueryCount from 'app/components/queryCount';
 import {t} from 'app/locale';
-import SentryTypes from 'app/sentryTypes';
+import {Fingerprint} from 'app/stores/groupingStore';
+import {Project} from 'app/types';
 
 import MergedItem from './mergedItem';
 import MergedToolbar from './mergedToolbar';
 
-class MergedList extends React.Component {
-  static propTypes = {
-    onUnmerge: PropTypes.func.isRequired,
-    onToggleCollapse: PropTypes.func.isRequired,
-    items: PropTypes.arrayOf(SentryTypes.Event),
-    pageLinks: PropTypes.string,
-    orgId: PropTypes.string.isRequired,
-    project: SentryTypes.Project.isRequired,
+type Props = {
+  // From GroupMergedView -> handleUnmerge
+  onUnmerge: () => {
```

Comment:
I'm not sure if I've done this signature correctly since I'm not used to Reflux.

Reply:
If you trace this down the component tree, you'll see that `mergedList` passes this to [mergedToolbar](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/views/organizationGroupDetails/groupMerged/mergedToolbar.jsx#L100) which eventually is passed to [<Confirm>](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/confirm.tsx#L49) which should be the func signature to use.

Reply:
From there I don't get much info to determine the signature:
https://github.com/getsentry/sentry/blob/b3c7104fc75dff77ce9115e0b27d12d69b7af7e8/src/sentry/static/sentry/app/views/organizationGroupDetails/groupMerged/mergedToolbar.jsx#L93-L103

From the top function I see references to `this.handleUnmerge` and `GroupingActions.toggleCollapseFingerprints`.
https://github.com/getsentry/sentry/blob/b3c7104fc75dff77ce9115e0b27d12d69b7af7e8/src/sentry/static/sentry/app/views/organizationGroupDetails/groupMerged/index.jsx#L119-L126

I see that `onMerge` calls `this.handleUnmerge()`:
https://github.com/getsentry/sentry/blob/b3c7104fc75dff77ce9115e0b27d12d69b7af7e8/src/sentry/static/sentry/app/views/organizationGroupDetails/groupMerged/index.jsx#L88-L95

This reads to me `() => ()` since there's no return inside. Nevertheless, there's a side-effect is called `GroupingActions.unmerge` which will eventually calls this (*I think*):
https://github.com/getsentry/sentry/blob/b3c7104fc75dff77ce9115e0b27d12d69b7af7e8/src/sentry/static/sentry/app/stores/groupingStore.tsx#L103-L108

In a similar fashion `onToggleCollapse={GroupingActions.toggleCollapseFingerprints}` will cause this code to be triggered:
https://github.com/getsentry/sentry/blob/b3c7104fc75dff77ce9115e0b27d12d69b7af7e8/src/sentry/static/sentry/app/stores/groupingStore.tsx#L118
https://github.com/getsentry/sentry/blob/b3c7104fc75dff77ce9115e0b27d12d69b7af7e8/src/sentry/static/sentry/app/stores/groupingStore.tsx#L483-L499

I'm assuming Reflux adds `on` in front of both functions (toggleCollapseFingerprints -> OnToggleCollapseFingerprints)

All being said, I think the signatures should be:
```
  onUnmerge: () => void
  onToggleCollapse: () => void;
```

What do you think?

Reply:
@armenzg Yep I think this is spot on